### PR TITLE
chore: Fix React `:nth-child` warnings

### PR DESCRIPTION
**Pull request by DafyddLlyr**
Been meaning to get around to this one for a while!

image

### CODE REVIEW

File: editor.planx.uk/src/ui/editor/RichTextInput.tsx [hunk 1]
```diff
@@ -96,15 +96,6 @@ export const RichContentContainer = styled(Box)(({ theme }) => ({
...
     },
-    // Styles for placeholder text, to match ui/Input.tsx
-    "& p.is-editor-empty:nth-child(1)::before": {
-      color: theme.palette.text.placeholder,
-      opacity: 1,
-      content: `attr(data-placeholder)`,
-      float: "left",
-      height: 0,
-      pointerEvents: "none",
-    },
```
Comment (DafyddLlyr) [inline editor.planx.uk/src/ui/editor/RichTextInput.tsx:107]: Removed as there's no reference to the class `is-editor-empty` elsewhere in the codebase, I believe this is redundant.

File: editor.planx.uk/src/ui/shared/InputRowLabel.tsx [hunk 2]
```diff
@@ -7,9 +7,6 @@ const Label = styled(Typography)(({ theme }) => ({
   flexGrow: 0,
   paddingRight: theme.spacing(2),
   alignSelf: "center",
-  "&:not(:nth-child(1))": {
-    paddingLeft: theme.spacing(2),
-  },
```
Comment (DafyddLlyr) [inline editor.planx.uk/src/ui/shared/InputRowLabel.tsx:12]: I can't see where the style applies, tried all usual tricks of solid red borders, hotpink background etc to locate when this should apply without any luck.

File: editor.planx.uk/src/ui/editor/RichTextInput.tsx (hunk 1 above)
Comment (RODO94) [reply editor.planx.uk/src/ui/editor/RichTextInput.tsx:107]: @DafyddLlyr I see that there's no longer placeholder text on the Editor Model:

Screenshot 2024-10-07 at 11 05 19

Wonder if this is linked to this change being removed?

**Review by RODO94 — COMMENTED** [review]
Thanks for picking this up! Absolute dream.

@DafyddLlyr one question still sitting there regarding placeholder text

File: editor.planx.uk/src/ui/editor/RichTextInput.tsx (hunk 1 above)
Comment (DafyddLlyr) [reply editor.planx.uk/src/ui/editor/RichTextInput.tsx:107]: Aha - I missed this somehow - taking a look thanks!

Comment (DafyddLlyr) [reply editor.planx.uk/src/ui/editor/RichTextInput.tsx]: Fixed please see 3af7314735349a77a817510f4d41a79653faa482

This was actually a class generated by tiptap ([docs](https://tiptap.dev/docs/editor/extensions/functionality/placeholder#additional-setup)).

File: editor.planx.uk/src/ui/editor/RichTextInput.tsx [hunk 3]
```diff
@@ -97,7 +97,8 @@ export const RichContentContainer = styled(Box)(({ theme }) => ({
       },
     },
     // Styles for placeholder text, to match ui/Input.tsx
-    "& p.is-editor-empty:nth-child(1)::before": {
+    // "& p.is-editor-empty:nth-child(1)::before": {
```
Comment (RODO94) [inline editor.planx.uk/src/ui/editor/RichTextInput.tsx:100]: nit: lingering comment?

**Review by RODO94 — APPROVED** [review]
Just one wee commented out line that I picked up but all looks good otherwise and everything still works for me

**Pull request merged**